Makefile: Upload wheels to Artifactory

samdbmg opened the pull request:
# Details
Amend the Makefile to set flags for uploading wheels to Artifactory (note also the URLs were added in c4b5b83, directly to main).

I've checked this by hacking the Makefile to allow pushing dev wheels and running `make upload-wheel` which successfully pushed to Artifactory: https://artifactory.virt.ch.bbc.co.uk/artifactory/webapp/#/artifacts/browse/tree/General/ap-python/httpio-bbc/1.0.2.dev2+g2245e37 

Flagging @j616 for review since you mentioned it last night, and I'd like to get your opinion on this one

# Pivotal Story
Story URL: https://www.pivotaltracker.com/story/show/180875963

# Related PRs
N/A

# Links to external test runs/working deployment
See above

# Submitter PR Checks
_(tick as appropriate)_

- [X] Added bbc/rd-apmm-cloudfit team as a reviewer
- [X] PR completes task/fixes bug
- [ ] Tests exercise code appropriately
- [ ] New features and API breaks are flagged in commit messages using magic strings
- [X] Repo maintainer is notified that a release is required
- [ ] Documentation updated (README, Confluence, Docstrings, API spec, Engineering Guide, etc.)
- [ ] Downstream repos have been checked for potential breaks & fixed as needed
- [ ] APIs/UIs/CLIs updated as required
- [X] PR added to Pivotal story
- [ ] Follow-up stories added to Pivotal
- [ ] Any temporary code/configuration removed (e.g. test deployment environment, temporary commontooling branch)
- [ ] Any pins against pre-releases have been removed

# Reviewer PR Checks
_(tick as appropriate)_

- [ ] PR completes task/fixes bug
- [ ] Tests exercise code appropriately
- [ ] Design makes sense, and fits with our current code base
- [ ] Code is easy to follow
- [ ] PR size is sensible
- [ ] Commit history is sensible and tidy

# Info on Cloudfit PRs
- The checks above are guidelines. They don't all have to be ticked, but they should all have been considered.
- For more details on how to asses PRs, see: https://github.com/bbc/rd-apmm-docs-ways-of-working/blob/master/workflow/PRChecklist.md

samdbmg committed:
Makefile: Upload wheels to Artifactory

j616 commented:
As discussed, it might be more resilient to use the following snippet to attempt setting of the defaults when running an `upload-wheels` command, rather than overriding in the repo's Makefile. If there is no access to "full" commontooling, this include will silently continue without importing
```make
ifeq ($(MAKECMDGOALS),upload-wheels)
-include commontooling/make/include/defaults.mk
endif
```

https://www.gnu.org/software/make/manual/html_node/Goals.html for info on `$(MAKECMDGOALS)`

samdbmg committed:
Use commontooling defaults for upload-wheel

Detects if the make target is upload-wheel and imports the full
commontooling defaults if available, to change the default repo to
Artifactory rather than PyPI.

samdbmg committed:
Regenerate static files

samdbmg commented:
Ok, how's this @j616 ?

j616 reviewed (approved):
LGTM
